src/Security/Authentication/Core/src/TicketSerializer.cs

```diff
@@ -212,6 +222,13 @@ protected virtual ClaimsIdentity ReadIdentity(BinaryReader reader)
             {
                 identity.Actor = ReadIdentity(reader);
             }
+             
+            // Determine whether the identity
+            // has a label.
+            if (reader.ReadBoolean())
```

Comment:
Have you tested with a ticket serialized using the previous version? Are you 100% sure this doesn't cause an `EndOfStreamException`?

Reply:
Yup, we need coverage for this. Maybe serialize something with the old version and add the raw bytes to a test.

Reply:
Yeah doesn't look like this works for back compat, since the identities are in the middle, I tested it by just removing the Write changes and seeing if the read changes would break, and it does.  Seems too dangerous to try and do this without a version bump

Reply:
Oh, right. I initially though this would be safe because it could be added at the end of an identity, but I forgot about multiple identities.

Reply:
Plan b. do we allow the deserialiser to read the v5 and v6 formats so there's at least some back compat?

Reply:
While we can certainly add that complexity, I have to ask if this is worth it at this point given how long we've never supported Label.  Seems like we should just either bump the version to 6 for label, or not make this change at this time.  I think bumping version and trying to support reading 5 and 6 is not worth the risk

Reply:
If we wanted to support back compat like that before, we always could have preserved older versioned deserializers around, which we clearly didn't

Reply:
Just bumping to 6 would cause too many interop issues to warrant adding Label support. The difference between 5 an 6 is small enough that we should be able to mitigate with one if check around the new label reading code. We should ask Barry if one-way compat is worth it.